src/event_timer.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""EventTimer for scheduling dispatch of juju event on regular intervals."""
+
+import subprocess  # nosec B404
+from pathlib import Path
+from typing import Optional, TypedDict
+
+from jinja2 import Environment, FileSystemLoader
+
+
+class TimerEnableError(Exception):
+    """Raised when unable to enable a event timer."""
+
+
+class TimerDisableError(Exception):
+    """Raised when unable to disable a event timer."""
+
+
+class EventConfig(TypedDict):
+    """Configuration used by service and timer templates."""
+
+    event: str
+    interval: float
+    jitter: float
+    timeout: float
+    unit: str
+
+
+class EventTimer:
+    """Manages the timer to emit juju events at regular intervals.
+
+    Attributes:
+        unit_name (str): Name of the juju unit to emit events to.
+    """
+
+    _systemd_path = Path("/etc/systemd/system")
+
+    def __init__(self, unit_name: str):
+        """Construct the timer manager.
+
+        Args:
+            unit_name: Name of the juju unit to emit events to.
+        """
+        self.unit_name = unit_name
+        self._jinja = Environment(loader=FileSystemLoader("templates"), autoescape=True)
+
+    def _render_event_template(self, template_type: str, event_name: str, context: EventConfig):
```

Comment:
Maybe it's because it reminds me of Flask's [render_template](https://flask.palletsprojects.com/en/2.3.x/api/#flask.render_template) but I wouldn't expect that a method with this name writes a file somewhere. What do you think?

Reply:
I believe it depends on the tool. Configuration management tools for example, render templates to files. In the context of web frameworks, I agree render does not always mean that a file will be written.